File: app/src/main/java/com/autoscout24/carfinder/ui/VehicleViewModelFactory.java
package com.autoscout24.carfinder.ui;

import android.arch.lifecycle.ViewModel;
import android.arch.lifecycle.ViewModelProvider;
import android.arch.lifecycle.ViewModelProviders;
import android.support.annotation.NonNull;

import com.autoscout24.carfinder.arch.data.VehicleRepository;

public class VehicleViewModelFactory extends ViewModelProvider.NewInstanceFactory {

    private VehicleRepository mRepository;

    public VehicleViewModelFactory(VehicleRepository repository) {
        this.mRepository = repository;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new VehicleListViewModel(mRepository);
    }
}

File: app/src/main/java/com/autoscout24/carfinder/ui/VehicleViewModelFactory.kt
package com.autoscout24.carfinder.ui

import android.arch.lifecycle.ViewModel
import android.arch.lifecycle.ViewModelProvider
import android.arch.lifecycle.ViewModelProviders

import com.autoscout24.carfinder.arch.data.VehicleRepository

class VehicleViewModelFactory(private val mRepository: VehicleRepository) : ViewModelProvider.NewInstanceFactory() {

    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        return VehicleListViewModel(mRepository) as T
    }
}
